File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) {
        int count = 0;
        double sum = 0;
        
        // initial sunset array of size k
        for( int i = 0; i < k;i++ ) 
            sum += arr[i];
        
        if( ( sum / k ) >= threshold )
            count++;
        
        // all other sub-arrays
        for( int i = k; i < arr.size(); i++ ) {
            sum = sum + arr[i] - arr[i - k];
            if( ( sum / k ) >= threshold )
                count++;
        }
        
        return count;
    }
};

File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.py
class Solution(object):
    def numOfSubarrays(self, arr, k, threshold):
        """
        :type arr: List[int]
        :type k: int
        :type threshold: int
        :rtype: int
        """
        winSum = sum(arr[:k])
        
        count = 0
        if float(winSum) / k >= threshold:
            count += 1
        
        for i in range(k, len(arr)):
            winSum = winSum + arr[i] - arr[i-k]
            if float(winSum) / k >= threshold:
                count += 1
        
        return count